src/compute/tests/integration_tests.rs

```diff
@@ -97,6 +109,283 @@ impl SingleChunkExecutor {
+            + 3 3.03 // binlog pos=2
+            + 4 4.04
+            + 5 5.05 // binlog pos=4
+            + 6 6.06",
+    );
+    let stream_chunk2 = StreamChunk::from_pretty(
+        " I F
+            + 6 10.08
+            + 199 40.5
+            + 978 72.6
+            + 134 41.7",
+    );
+
+    // The first barrier
+    let curr_epoch = 11;
+    tx.push_barrier(curr_epoch, false);
+
+    // assert barrier is forwarded to mview
+    assert!(matches!(
+        materialize.next().await.unwrap()?,
+        Message::Barrier(Barrier {
+            epoch,
+            mutation: None,
+            ..
+        }) if epoch.curr == curr_epoch
+    ));
+
+    // start the stream pipeline src -> backfill -> mview
+    let mview_handle = tokio::spawn(consume_message_stream(materialize));
+
+    // ingest data and barrier
+    let interval = Duration::from_millis(10);
+    tx.push_chunk(stream_chunk1);
+
+    tokio::time::sleep(interval).await;
+    tx.push_barrier(curr_epoch + 1, false);
+
+    tx.push_chunk(stream_chunk2);
+
+    tokio::time::sleep(interval).await;
+    tx.push_barrier(curr_epoch + 2, false);
+
+    tokio::time::sleep(interval).await;
+    tx.push_barrier(curr_epoch + 3, true);
+
+    // scan the final result of the mv table
+    let column_descs = vec![
+        ColumnDesc::unnamed(ColumnId::from(0), schema[0].data_type.clone()),
+        ColumnDesc::unnamed(ColumnId::from(1), schema[1].data_type.clone()),
+    ];
+    let value_indices = (0..column_descs.len()).collect_vec();
+    // Since we have not polled `Materialize`, we cannot scan anything from this table
+    let table = StorageTable::for_test(
+        memory_state_store.clone(),
+        table_id,
+        column_descs.clone(),
+        vec![OrderType::ascending()],
+        vec![0],
+        value_indices,
+    );
+
+    let scan = Box::new(RowSeqScanExecutor::new(
+        table.clone(),
+        vec![ScanRange::full()],
+        true,
+        to_committed_batch_query_epoch(u64::MAX),
+        1024,
+        "RowSeqExecutor2".to_string(),
+        None,
+    ));
+    let mut stream = scan.execute();
+    while let Some(message) = stream.next().await {
+        println!("[scan] chunk: {:#?}", message.unwrap());
+    }
+
+    mview_handle.await.unwrap()?;
+
+    Ok(())
+}
+
+async fn consume_message_stream(mut stream: BoxedMessageStream) -> StreamResult<()> {
+    while let Some(message) = stream.next().await {
+        let message = message?;
+        match message {
+                Message::Watermark(_) => {
+                    break;
+                }
+                Message::Chunk(c) => {
+                    println!("[mv] chunk: {:#?}", c);
+                }
+                Message::Barrier(b) => {
+                    if let Some(m) = b.mutation && matches!(*m, Mutation::Stop(_)) {
+                        println!("encounter stop barrier");
+                        break
+                    }
+                }
+            }
+    }
+    Ok(())
+}
+
```

Comment:
Btw for testing the updates processing branch of mview-on-mview backfill, we have this test `test_backfill_with_upstream_and_snapshot_read`, which does concurrent updates. You may copy it to test for cdc backfill too.

I guess source may not be supported in deterministic testing, but perhaps you can just replicate it in a shell script.